src/tests/loadtxt/test_loadtxt_qp.f90

```diff
@@ -0,0 +1,38 @@
+program test_loadtxt_qp
+use iso_fortran_env, only: sp=>real32, dp=>real64, qp=>real128
+use stdlib_experimental_io, only: loadtxt
+implicit none
+
+real(qp), allocatable :: q(:, :)
+
+call loadtxt("array4.dat", q)
+call print_array(q)
+
+contains
+
+subroutine print_array(a)
+class(*),intent(in) :: a(:, :)
+integer :: i
+print *, "Array, shape=(", size(a, 1), ",", size(a, 2), ")"
+
+ select type(a)
+  type is(real(sp))
+   do i = 1, size(a, 1)
+    print *, a(i, :)
+   end do
+  type is(real(dp))
+   do i = 1, size(a, 1)
+    print *, a(i, :)
+   end do
```

Comment:
I would deleted `type is (real(sp))` and `type is(real(dp))` to avoid confusion with the name of the program.

Reply:
Done in 84975608064f4fdaa1160ee79bc45508b4d771e9.